File: src/api/types/dashboard/bot.rs
#[derive(Debug, Serialize)]
pub struct BotVerifierStatusResponse {
    pub(crate) generated_at: i64,
    pub(crate) providers: Vec<BotVerifierProviderStatusResponse>,
}

#[derive(Debug, Serialize)]
pub struct BotVerifierProviderStatusResponse {
    pub(crate) provider: String,
    pub(crate) range_count: usize,
    pub(crate) last_refresh_at: Option<i64>,
    pub(crate) last_success_at: Option<i64>,
    pub(crate) last_error: Option<String>,
    pub(crate) status: String,
}

#[derive(Debug, Serialize)]
pub struct BotInsightsResponse {
    pub(crate) generated_at: i64,
    pub(crate) window_start: i64,
    pub(crate) total_bot_events: u64,
    pub(crate) by_trust_class: Vec<AiAuditCountItem>,
    pub(crate) top_bot_names: Vec<AiAuditCountItem>,
    pub(crate) top_mismatch_ips: Vec<AiAuditCountItem>,
    pub(crate) top_routes: Vec<AiAuditCountItem>,
}
use serde::Serialize;

use super::AiAuditCountItem;

File: dome-vue/src/features/dashboard/types/dashboard/bot.ts
import type { AiAuditCountItem } from './aiAudit'

export interface BotVerifierStatusResponse {
  generated_at: number
  providers: BotVerifierProviderStatus[]
}

export interface BotVerifierProviderStatus {
  provider: string
  range_count: number
  last_refresh_at: number | null
  last_success_at: number | null
  last_error: string | null
  status: string
}

export interface BotInsightsResponse {
  generated_at: number
  window_start: number
  total_bot_events: number
  by_trust_class: AiAuditCountItem[]
  top_bot_names: AiAuditCountItem[]
  top_mismatch_ips: AiAuditCountItem[]
  top_routes: AiAuditCountItem[]
}
